core/src/main/java/com/oath/oak/OakAttachedReadBuffer.java

```diff
@@ -37,48 +37,57 @@ public int capacity() {
 
     @Override
     public ByteOrder order() {
-        return buffer.order();
+        return readBuffer.order();
     }
 
     @Override
     public byte get(int index) {
-        return getDataByteBuffer().get(getDataOffset(index));
+        return readBuffer.get(getDataOffset(index));
     }
 
     @Override
     public char getChar(int index) {
-        return getDataByteBuffer().getChar(getDataOffset(index));
+        return readBuffer.getChar(getDataOffset(index));
     }
 
     @Override
     public short getShort(int index) {
-        return getDataByteBuffer().getShort(getDataOffset(index));
+        return readBuffer.getShort(getDataOffset(index));
     }
 
     @Override
     public int getInt(int index) {
-        return getDataByteBuffer().getInt(getDataOffset(index));
+        return readBuffer.getInt(getDataOffset(index));
     }
 
     @Override
     public long getLong(int index) {
-        return getDataByteBuffer().getLong(getDataOffset(index));
+        return readBuffer.getLong(getDataOffset(index));
     }
 
     @Override
     public float getFloat(int index) {
-        return getDataByteBuffer().getFloat(getDataOffset(index));
+        return readBuffer.getFloat(getDataOffset(index));
     }
 
     @Override
     public double getDouble(int index) {
-        return getDataByteBuffer().getDouble(getDataOffset(index));
+        return readBuffer.getDouble(getDataOffset(index));
     }
 
-    /*-------------- OakUnsafeDirectBuffer --------------*/
-
-    @Override
-    public ByteBuffer getByteBuffer() {
-        return getDuplicatedReadByteBuffer();
+    /*-------------- Internal --------------*/
+
+    /**
+     * This method should be used internally by Oak to write to the buffer.
+     * It creates a new temporary writable buffer and disable it once the write is done.
+     * @param writer a method that writes to a writable buffer
+     */
+    void write(Consumer<OakWriteBuffer> writer) {
```

Comment:
Is it that try/catch that you saw worth eliminating? In general write of the read buffer looks very awkward to me...

Reply:
Outdated.